core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -131,6 +135,17 @@ case class SparkListenerApplicationEnd(time: Long) extends SparkListenerEvent
 private[spark] case class SparkListenerLogStart(sparkVersion: String) extends SparkListenerEvent
 
 /**
+ * Interface for registering listeners defined in other modules like SQL, which are used to
+ * rebuild the history UI.
+ */
+trait SparkListenerRegister {
```

Comment:
I know it's a little annoying to rename things, but I don't like this name very much... how about `SparkHistoryListenerFactory` or something along those lines?

Also, this trait should probably be `private[spark]`.